packages/plugin-board/src/board.js

```diff
@@ -0,0 +1,444 @@
+   * Gets all Content from a Channel
+   * It will make multiple GET requests if contents length are greater than
+   * MAX_ALLOW_INPUT_SIZE, the number is currently determined and hard-coded
+   * by the backend
+   * @memberof Board.BoardService
+   * @param  {Board~Channel} channel
+   * @returns {Promise<Array>} Resolves with an Array of {@link Board~Content} objects.
+   */
+  getAllContent(channel) {
+    let contents = [];
+    let hasMoreContents = false;
+
+    const loop = () => {
+      if (!hasMoreContents) {
+        return Promise.resolve(contents);
+      }
+      const oldestContentTime = last(contents).createdTime;
+
+      return this._getPageOfContents(channel, {sinceDate: oldestContentTime})
+        .then((res) => {
+
+          if (res.length >= MAX_ALLOWED_INPUT_SIZE) {
+            hasMoreContents = true;
+          }
+          else {
+            hasMoreContents = false;
+          }
+
+          // for the 2nd+ request, the result will include the last item of
+          // the first request, so we'll discard it here
+          res.shift();
+          contents = contents.concat(res);
+          return contents;
+        })
+        .then(loop.bind(this));
+    };
+
+    return this._getPageOfContents(channel)
+      .then((res) => {
+        contents = res;
+
+        if (contents.length >= MAX_ALLOWED_INPUT_SIZE) {
+          hasMoreContents = true;
+        }
+        return contents;
+      })
+      .then(loop.bind(this));
+  },
+
+  /**
+   * Gets a Channel
+   * @memberof Board.BoardService
+   * @param  {Board~Channel} channel
+   * @returns {Promise<Board~Channel>}
+   */
+  getChannel(channel) {
+    return this.spark.request({
+      method: `GET`,
+      uri: channel.channelUrl
+    })
+      .then((res) => res.body);
+  },
+
+  /**
+   * Gets Channels
+   * @memberof Board.BoardService
+   * @param {Object} options
+   * @param {number} options.limit Max number of activities to return
+   * @returns {Promise<Page<Board~Channel>>} Resolves with an array of Channel items
+   */
+  getChannels(options) {
+    options = options || {};
+
+    if (!options.conversationId) {
+      return Promise.reject(new Error(`\`conversationId\` is required`));
+    }
+
+    const params = {
+      api: `board`,
+      resource: `/channels`,
+      qs: {}
+    };
+    assign(params.qs, pick(options, `channelsLimit`, `conversationId`));
+
+    return this.request(params)
+      .then((res) => new Page(res, this.spark));
+  },
+
+  /**
+   * Pings persistence
+   * @memberof Board.BoardService
+   * @returns {Promise<Object>} ping response body
+   */
+  ping() {
+    return this.spark.request({
+      method: `GET`,
+      api: `board`,
+      resource: `/ping`
+    })
+      .then((res) => res.body);
+  },
+
+  processActivityEvent(message) {
+    let decryptionPromise;
+
+    if (message.contentType === `FILE`) {
+      decryptionPromise = this.decryptSingleFileContent(message.payload, message.envelope.encryptionKeyUrl);
```

Comment:
note that the envelope is no longer emitted in position 2 for autowired event handlers (I think there's a bug where i intended to move it to position 1 like i did with regular event handlers). if the encryptionKeyUrl is important, you may need to adjust _onmessage in plugin-mercury as well as adjust all process*Event methods accordingly.